File: scratch/test_pooler.js
const { Client } = require('pg');
const dns = require('dns');
dns.setDefaultResultOrder('ipv4first');

async function main() {
  console.log("Testing connection via Node.js with pooler...");
  const client = new Client({
    user: 'postgres.xernedkpgdrvjokokdoa',
    password: 'akshitsha84',
    host: 'aws-0-ap-northeast-2.pooler.supabase.com',
    port: 6543,
    database: 'postgres',
    ssl: { rejectUnauthorized: false }
  });

  try {
    await client.connect();
    console.log("🎉 SUCCESS! Connected successfully via Node.js pooler!");
    const res = await client.query("SELECT COUNT(*) FROM public.students;");
    console.log("Student count:", res.rows[0].count);
  } catch (err) {
    console.error("Connection failed:", err.message);
  } finally {
    await client.end();
  }
}

main();

File: test_pooler.py
import psycopg2
import sys

def main():
    regions = [
        "ap-northeast-2",  # Seoul (resolved IP location)
        "ap-south-1",      # Mumbai
        "ap-southeast-1",  # Singapore
        "ap-northeast-1",  # Tokyo
        "us-east-1",       # N. Virginia
    ]
    
    password = "akshitsha84"
    project_ref = "xernedkpgdrvjokokdoa"
    
    for region in regions:
        host = f"aws-0-{region}.pooler.supabase.com"
        for user in [f"postgres.{project_ref}", "postgres"]:
            db_uri = f"postgresql://{user}:{password}@{host}:6543/postgres"
            print(f"Testing {region} with user={user}...")
            try:
                conn = psycopg2.connect(db_uri, connect_timeout=3)
                print(f"SUCCESS! Connected with user={user} on {region}!")
                conn.close()
                sys.exit(0)
            except Exception as e:
                err = str(e).strip()
                print(f"  Failed: {err[:80]}")

if __name__ == "__main__":
    main()
